piAIOComm.c

```diff
@@ -266,80 +269,53 @@ INT32U piAIOComm_Config(uint8_t i8uAddress, uint16_t i16uNumEntries, SEntryInfo
 
 u32 piAIOComm_Init(u8 devnum)
 {
-	u8 addr = RevPiDevice_getDev(devnum)->i8uAddress;
+	SAioConfig *config;
+	u8 *snd_buf;
+	u8 addr;
 	int ret;
-	SIOGeneric sResponse_l;
-	INT8U i, len_l;
+	int i;
+
+	addr = RevPiDevice_getDev(devnum)->i8uAddress;
 
-	pr_info_aio("piAIOComm_Init %d of %d  addr %d\n", devnum, i8uConfigured_s,
-		    RevPiDevice_getDev(devnum)->i8uAddress);
+	pr_info_aio("piAIOComm_Init %d of %d  addr %d\n", devnum,
+		    i8uConfigured_s, addr);
 
 	for (i = 0; i < i8uConfigured_s; i++) {
-		if (aioConfig_s[i].uHeader.sHeaderTyp1.bitAddress == addr) {
-			pr_info_aio("piAIOComm_Init send configIn1\n");
-			ret = piIoComm_send((INT8U *) & aioIn1Config_s[i], sizeof(SAioInConfig));
-			if (ret == 0) {
-				len_l = 0;	// empty config telegram
-
-				ret = piIoComm_recv((INT8U *) & sResponse_l, IOPROTOCOL_HEADER_LENGTH + len_l + 1);
-				if (ret > 0) {
-					if (piIoComm_response_valid(&sResponse_l, addr, len_l)) {
-						pr_info_aio("piAIOComm_Init done configIn1\n");
-						// success -> continue
-					} else {
-						return 1;	// wrong crc
-					}
-				} else {
-					return 2;	// no response
-				}
-			} else {
-				return 3;	// could not send
-			}
-
-			pr_info_aio("piAIOComm_Init send configIn2\n");
-			ret = piIoComm_send((INT8U *) & aioIn2Config_s[i], sizeof(SAioInConfig));
-			if (ret == 0) {
-				len_l = 0;	// empty config telegram
-
-				ret = piIoComm_recv((INT8U *) & sResponse_l, IOPROTOCOL_HEADER_LENGTH + len_l + 1);
-				if (ret > 0) {
-					if (piIoComm_response_valid(&sResponse_l, addr, len_l)) {
-						pr_info_aio("piAIOComm_Init done configIn2\n");
-						// success -> continue
-					} else {
-						return 1;	// wrong crc
-					}
-				} else {
-					return 2;	// no response
-				}
-			} else {
-				return 3;	// could not send
-			}
-
-
-			pr_info_aio("piAIOComm_Init send config\n");
-			ret = piIoComm_send((INT8U *) & aioConfig_s[i], sizeof(SAioConfig));
-			if (ret == 0) {
-				len_l = 0;	// empty config telegram
-
-				ret = piIoComm_recv((INT8U *) & sResponse_l, IOPROTOCOL_HEADER_LENGTH + len_l + 1);
-				if (ret > 0) {
-					if (piIoComm_response_valid(&sResponse_l, addr, len_l)) {
-						pr_info_aio("piAIOComm_Init done config\n");
-						return 0;	// success -> leave the function
-					} else {
-						return 1;	// wrong crc
-					}
-				} else {
-					return 2;	// no response
-				}
-			} else {
-				return 3;	// could not send
-			}
-		}
+		config = &aioConfig_s[i];
+		if (config->uHeader.sHeaderTyp1.bitAddress == addr)
+			break;
 	}
 
-	return 4;		// unknown device
+	if (i == i8uConfigured_s)
+		return 4; // unknown device
```

Comment:
Good idea to add this check and thus make it possible to reduce indentation below by 2 tabs.